### ISSUE TITLE
How to get the firebase support for flutter in tizen?

### ISSUE THREAD

**Issue by RaghuMudem** [flutter-tizen/flutter-tizen#229]
We have a working flutter application, which have the firebase support from the libraries mentioned below. We run the same application from Tizen studio to smart tv simulator. And experiencing the below mentioned error logs. We are more interested to build our application with Tizen. 
  
Any suggestions to resolve the dependency errors.
Or Help to convert flutter application into tizen-flutter application.
Or Guidance to use the existed https://pub.dev/ packages into tizen smart tv application.

**Flutter Firebase libraries list:**
-   firebase_core: ^1.5.0
-   firebase_core_web: ^1.0.2
-   firebase_auth: ^3.0.2
-   firebase_analytics: ^8.3.0
-   firebase_storage: ^10.0.2
-   cloud_firestore: ^2.5.0
-   cloud_functions: ^3.0.1
-   firebase_crashlytics: ^2.2.0
-   firebase_messaging: ^10.0.5
-   firebase_messaging_web: ^2.0.4  
-----------------------------
**Error logs:**
C:\Users\lans497\Documents\GitHub\flutter_pca>flutter-tizen run
Multiple devices found:
Tizen T-samsung-6.0-x86 (mobile) • emulator-26101        • flutter-tester • Tizen 6.0 (emulator)
Tizen UE43AU7172UXXH (mobile)    • 192.168.123.134:26101 • flutter-tester • Tizen 6.0
[1]: Tizen T-samsung-6.0-x86 (emulator-26101)
[2]: Tizen UE43AU7172UXXH (192.168.123.134:26101)
Please choose one (To quit, press "q/Q"): 1
device_info_plus_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
image_picker_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
package_info_plus_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
path_provider_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
permission_handler_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
share_plus_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
shared_preferences_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
url_launcher_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
integration_test_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
webview_flutter_tizen is available on pub.dev. Did you forget to add to pubspec.yaml?
Using hardware rendering with device Tizen T-samsung-6.0-x86. If you notice graphics artifacts, consider enabling
software rendering with "--enable-software-rendering".
Launching lib\main.dart on Tizen T-samsung-6.0-x86 in debug mode...
The TizenEmul profile is used for signing.
Building a Tizen application in debug mode...                      80.3s
√ Built build\tizen\tpk\com.example.my_samsung_app-210818.10.56.tpk (33.3MB).
Installing build\tizen\tpk\com.example.my_samsung_app-210818.10.56.tpk...        19.3s
I/ConsoleMessage( 3792): Utils.cs: ParseEngineArgs(30) > Enabled: --enable-dart-profiling

I/ConsoleMessage( 3792): Utils.cs: ParseEngineArgs(30) > Enabled: --enable-checked-mode

E/ConsoleMessage( 3792): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)

E/ConsoleMessage( 3792): #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156:7)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792): #1      MethodChannel.invokeListMethod (package:flutter/src/services/platform_channel.dart:344:35)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792): #2      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:30:23)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792): #3      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:77:7)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792): #4      Firebase.initializeApp (package:firebase_core/src/firebase.dart:42:31)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792): #5      main (package:pca_app/main.dart:69:3)

E/ConsoleMessage( 3792): <asynchronous suspension>

E/ConsoleMessage( 3792):

Syncing files to device Tizen T-samsung-6.0-x86...                 319ms

Flutter run key commands.
r Hot reload.
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

Running with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

An Observatory debugger and profiler on Tizen T-samsung-6.0-x86 is available at: http://127.0.0.1:59007/ceggtT2_fNM=/
The Flutter DevTools debugger and profiler on Tizen T-samsung-6.0-x86 is available at:
http://127.0.0.1:9109?uri=http%3A%2F%2F127.0.0.1%3A59007%2FceggtT2_fNM%3D%2F

**Comment by swift-kim** [comment]
Unfortunately, Flutter Firebase plugins are not available for Tizen as mentioned in [this wiki page](https://github.com/flutter-tizen/flutter-tizen/wiki/Limitations#plugins).

FlutterFire packages are not just packages, but they are **plugins** that should be implemented in each platform's native language (e.g. Java for Android). Some major platforms, such as Android and web, have their own native Firebase libraries maintained by the Firebase team, and Firebase Flutter plugins are created from those libraries again by the Firebase team. Tizen lacks both, and it will be a massive amount of work (or almost impossible) if we want to implement them without support from Google. Also if you want to implement authentication in your app, you will probably want to use the `firebase_auth` package which indirectly depends on the `google_sign_in` package. `google_sign_in` is also a plugin, and there's no Tizen implementation for it (although at least there's a [plan](https://github.com/flutter-tizen/plugins/issues/151)).

There are still other options, such as [firedart](https://github.com/cachapa/firedart) or using Firebase REST APIs directly. However, they only support limited features of Firebase APIs, and also it will require additional work for you.

In short, Flutter apps that use Firebase plugins cannot be ported to Tizen easily, although it's technically possible.

@haesik If you're interested, could you estimate the exact amount of work required to add Tizen support for Firebase plugins? (Please talk to me when I'm not out of office.)

**Comment by RaghuMudem** [comment]
Hi, @swift-kim thanks for your reply.

Do you or your team, has any plans for this in future? If yes, then can we know the possibility timeline for this to complete. 
We plan to release our product in Tizen version which is blocking by the firebase unavailability.

**Comment by swift-kim** [comment]
@RaghuMudem To be clear, we don't have enough resources to implement and maintain FlutterFire packages for Tizen, so it's unlikely they will be ever available. Sorry for the inconvenience.

**Comment by swift-kim** [comment]
News: The Invertase and Canonical developers are working hard on the FlutterFire Dart SDK (although it's still in a very early stage).

https://invertase.io/blog/announcing-flutterfire-desktop

So now there's a chance Tizen will get Firebase support based on their work in the future. However, only limited features of FlutterFire may be supported due to the fundamental limitations of the Firebase REST APIs and Google OAuth API.

**Comment by swift-kim** [comment]
Closing in favor of https://github.com/flutter-tizen/flutter-tizen/issues/353.
